models/userStatus.js

```diff
@@ -327,101 +327,111 @@ const updateStatusOnTaskCompletion = async (userId) => {
   }
 };
 
-const massUpdateIdleUsers = async (users) => {
+const batchUpdateUsersStatus = async (users) => {
   const currentTimeStamp = new Date().getTime();
   const batch = firestore.batch();
-  let usersWithStatusUnchanged = 0;
-
-  await Promise.all(
-    users.map(async (userId) => {
-      let latestStatusData;
-      try {
-        latestStatusData = await getUserStatus(userId);
-      } catch (error) {
-        usersWithStatusUnchanged++;
-        return batch;
-      }
-      const { id, userStatusExists, data } = latestStatusData;
+  const summary = {
+    totalUsers: users.length,
+    totalUnprocessedUsers: 0,
+    totalOnboardingUsersAltered: 0,
+    totalOnboardingUsersUnAltered: 0,
+    totalActiveUsersAltered: 0,
+    totalActiveUsersUnAltered: 0,
+    totalIdleUsersAltered: 0,
+    totalIdleUsersUnAltered: 0,
+  };
 
-      if (!userStatusExists || !data?.currentStatus) {
-        const newUserStatusRef = userStatusModel.doc();
-        const newUserStatusData = {
-          userId,
-          currentStatus: {
-            state: userState.IDLE,
-            message: "",
-            from: currentTimeStamp,
-            until: "",
-            updatedAt: currentTimeStamp,
-          },
-        };
-        batch.set(newUserStatusRef, newUserStatusData);
-      } else {
-        const {
-          currentStatus: { state, until },
-        } = data;
+  for (const { userId, expectedState } of users) {
+    let latestStatusData;
+    try {
+      latestStatusData = await getUserStatus(userId);
+    } catch (error) {
+      summary.totalUnprocessedUsers++;
+      return batch;
```

Comment:
Why not return error? Why is batch being returned here?

Reply:
Yes this was not needed. I removed it now. earlier I was using map so I needed return statement. Now I am no longer using map so I don't need return. Hence removed.

Reply:
Why are we continuing also?

Reply:
I understood. Resolving this.

Reply:
Thanks